riscv.md

```diff
@@ -416,6 +417,44 @@ typedef struct {
 } tls_index;
 ```
 
+## <a name=program-headers></a> Program Header Table
+
+The following table lists additional processor-specific segment types:
+
+Name                      | Value      | Description
+--------------------------|------------|-------------------------------------
+PT\_RISCV\_MEMTAG\_CHERI  | 0x7fffffff | Capability memory tags
+
+`PT_RISCV_MEMTAG_CHERI` segments (if present) hold capability memory tags
+for a particular memory range. They are defined for core dump files of
+type `ET_CORE`.  The program header for these segments is defined as:
+
+* `p_type`: `PT_RISCV_MEMTAG_CHERI`
+
+* `p_flags`: bitmask of capability tag MMU permissions
+
+* `p_offset`: segment file offset
+
+* `p_vaddr`: segment virtual address, must be page-aligned
+
+* `p_paddr`: 0
+
+* `p_filesz`: segment size in file, calculated as `p_memsz / (8 *
+  sizeof(void * __capability)` (eight 1-bit tags cover 8 capabilities of
+  memory)
+
+* `p_memsz`: segment size in memory, must be a multiple of the page size
```

Comment:
Is so you can compute the stride?

Reply:
Not really (the discussion we had when defining this for ARM is that the stride should be assumed for each specific PT_<arch>_MEMTAG_<type> vs having some kind of generic PT_MEMTAG).  However, the value of `p_memsz` here is sensible and also permits building tables mapping file ranges to virtual address ranges in debuggers.